components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/utils/SecondaryUserStoreConfigurationUtil.java

```diff
@@ -790,4 +862,21 @@ public static boolean isUserStoreRepositorySeparationEnabled() {
         // https://github.com/wso2/product-is/issues/5673
         return false;
     }
+
+    /**
+     * Wrap UserStoreClientException in a IdentityUserStoreClientException.
+     *
+     * @param defaultMessage Default error message.
+     * @param e UserStoreClientException.
+     * @return new IdentityUserStoreClientException.
+     */
+    public static IdentityUserStoreClientException buildIdentityUserStoreClientException(String defaultMessage,
+                                                                                         UserStoreClientException e) {
+
+        String errorMessage = defaultMessage;
+        if (e.getMessage() != null) {
+            errorMessage = e.getMessage();
+        }
+        return new IdentityUserStoreClientException(errorMessage, e);
```

Comment:
Hope these errors are handled through CXF errors  and printed properly in case if debug is turned on. Unless these error messages can go unnoticed even if debug is turned on.